ui-tests/cypress/support/pageObjects/pages/designer/widgetconfigs/ContentWidgetConfigPage.js

```diff
@@ -30,11 +32,35 @@ export class ContentListSelectModal extends DialogContent {
 }
 
 export default class ContentWidgetConfigPage extends WidgetConfigPage {
+
+  addButtonArea = `${htmlElements.div}.SingleContentConfigFormBody__addButtons`;
+  buttonClass = `${htmlElements.button}.btn.btn-primary`;
+  buttonDropdown = `${htmlElements.div}.dropdown.btn-group-primary`;
+  modelIdSelect = `${htmlElements.select}[name="modelId"]`;
+
+  getAddButtonsArea() {
+    return this.getInnerPanel().find(this.addButtonArea);
+  }
   
   getAddContentButton() {
-    // TODO - amend test id attributes for the buttons in appbuilder to avoid using `contains` method
-    return this.getInnerPanel().find('button.btn.btn-primary')
-      .contains(/^Add existing content$/);
+    return this.getAddButtonsArea().children(this.buttonClass);
+  }
+
+  getAddNewButtonDropdown() {
+    return this.getAddButtonsArea().children(this.buttonDropdown);
+  }
+
+  getButtonAddByContentTypeName(ctype) {
+    return this.getAddNewButtonDropdown()
+      .children(htmlElements.ul)
+      .children(htmlElements.li)
+      .contains(ctype);
```

Comment:
Should not use `contains` to identify elements in the page

Reply:
@g-di-martino I am forced to use `contains` due to the menu items being dynamically populated with list of content type names. the content type I wanted to click could be in different order inside the dropdown menu list.

Reply:
add a `FIXME` on top of it, will find a way

Reply:
done